daemon/delete.go

```diff
@@ -55,7 +57,19 @@ func (daemon *Daemon) ContainerRm(job *engine.Job) engine.Status {
 		if container.IsRunning() {
 			if forceRemove {
 				if err := container.Kill(); err != nil {
-					return job.Errorf("Could not kill running container, cannot remove - %v", err)
+					if err == execdriver.ErrProcessNotFound {
+						// This is a situation in which the container PID 1 has died
+						// but has yet to be reaped or recognized by Docker as being gone.
+						// wait for the container to stop
+						if _, err := container.WaitStop(10 * time.Second); err != nil {
+							return job.Errorf("Failed to wait for killed container to stop, cannot remove - %v", err)
+						}
+						if container.IsRunning() {
+							return job.Errorf("Failed to kill container, cannot remove - still running")
+						}
+					} else {
```

Comment:
I'm not sure, but I am personally prefer to use `early returns` without `else` clause.
Like
```
if err != execdriver.ErrProcessNotFound {
    return job.Errorf("Could not kill running container, cannot remove - %v", err)
}
```